### PR TITLE
Ability to add MessageTransforms to the GroupChat's Select Speaker nested chat (speaker_selection_method='auto')

### PR DESCRIPTION
**Pull request by marklysze**
This provides the ability to add MessageTransforms to the nested select speaker chat that occurs in a GroupChat when the speaker selection method is 'auto'. The MessageTransforms apply only to the messages within that nested select speaker chat (and not to the broader GroupChat chat).

The GroupChat constructor has a new, optional, parameter called `select_speaker_auto_message_transforms` that takes in a `MessageTransforms` object, or `None`.

Within GroupChat, the provided MessageTransforms object will be assigned to `self._auto_message_transforms` and used within `_auto_select_speaker` and `a_auto_select_speaker`.

The nested select speaker chat has two agents, `checking_agent` and `speaker_selection_agent` and the MessageTransforms are added to the `speaker_selection_agent` as that is the agent that utilises the LLM.

## Why are these changes needed?

These changes enable greater control and flexibility of messages for speaker selection. Particularly useful for non-OpenAI models with limited context / reasoning ability.

## Checks

- [X] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [X] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

(2719)

### CODE REVIEW

**Comment by qingyun-wu** [comment]
@marklysze fix the code formating issue? Let me know if you need help!

**Comment by marklysze** [comment]
> @marklysze fix the code formating issue? Let me know if you need help!

Okie dokie, should be right to go!

File: website/docs/topics/groupchat/transform_messages_speaker_selection.ipynb [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    "    select_speaker_transform_messages=select_speaker_transforms,\n",
+    ")"
+   ]
+  }
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "Custom Speaker Selection Function",
+   "tags": [
+    "orchestration",
```
Comment (sonichi) [inline website/docs/topics/groupchat/transform_messages_speaker_selection.ipynb:222]:
```suggestion
    "orchestration",
    "long context handling",
```

**Review by sonichi — APPROVED** [review]

**Comment by marklysze** [comment]
Just a note - sorry, I accidentally had a local testing file in the commit. I've deleted it.

**Pull request merged**